crates/oxc_linter/src/rules/no_empty.rs

```diff
@@ -0,0 +1,78 @@
+use oxc_ast::{AstKind, Span};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+
+use crate::{context::LintContext, rule::Rule};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-empty): Disallow empty block statements")]
+#[diagnostic(severity(warning), help("Add comments inside {0} statement"))]
+struct NoEmptyDiagnostic(&'static str, #[label("Empty {0} statement")] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoEmpty;
```

Comment:
TODO: macro this so we get nice documentation out of this. Checkout how Rome / Ruff / Clippy does it.

Reply:
TODO: allowEmptyCatch